File: em Vídeo/POO [60 Horas]/PHP POO [40 Horas]/Aula 05/ContaBanco.php
<?php
    class ContaBanco {
        public $numConta;
        protected $tipo;
        private $dono;
        private $saldo;
        private $status;

        public function __construct() {
            $this->setSaldo(0);
            $this->setStatus(false);
        }

        public function getNumConta() {
            return $this->numConta;
        }

        public function setNumConta($numConta) {
            $this->numConta = $numConta;
        }

        public function getTipo() {
            return $this->tipo;
        }

        public function setTipo($tipo) {
            $this->tipo = $tipo;
        }

        public function getDono() {
            return $this->dono;
        }

        public function setDono($dono) {
            $this->dono = $dono;
        }

        public function getSaldo() {
            return $this->saldo;
        }

        public function setSaldo($saldo) {
            $this->saldo = $saldo;
        }

        public function isStatus() {
            return $this->status;
        }

        public function setStatus($status) {
            $this->status = $status;
        }

        public function abrirConta($t) {
            $this->setTipo($t);
            $this->setStatus(true);
            if ($t == "CC") {
                $this->setSaldo(50);
            } elseif ($t == "CP") {
                $this->setSaldo(150);
            }
        }

        public function fecharConta() {
            if ($this->getSaldo() > 0) {
                print("Conta com dinheiro\n");
            } elseif ($this->getSaldo() < 0) {
                print("Conta em débito\n");
            } else {
                $this->setStatus(false);
                print("Conta de ".$this->getDono()." fechada com sucesso\n");
            }
        }

        public function depositar($v) {
            if ($this->isStatus()) {
                $this->setSaldo($this->getSaldo() + $v);
                print("Depósito de R$$v autorizado na conta de ".$this->getDono()."\n");
            } else {
                print("Impossível depositar\n");
            }
        }

        public function sacar($v) {
            if ($this->isStatus()) {
                if ($this->getSaldo() >= $v) {
                    $this->setSaldo($this->getSaldo() - $v);
                    print("Saque de R$$v autorizado na conta de ".$this->getDono()."\n");
                } else {
                    print("Saldo insuficiente\n");
                }
            } else {
                print("Impossível sacar\n");
            }
        }

        public function pagarMensal() {
            if ($this->getTipo() == "CC") {
                $v = 12;
            } else if ($this->getTipo() == "CP") {
                $v = 20;
            }
            if ($this->isStatus()) {
                if ($this->getSaldo() >= $v) {
                    $this->setSaldo($this->getSaldo() - $v);
                    print("Mensalidade de R$$v debitada na conta de ".$this->getDono()."\n");
                } else {
                    print("Saldo insuficiente\n");
                }
            } else {
                print("Impossível pagar.\n");
            }
        }

    }
?>

File: em Vídeo/POO [60 Horas]/Java POO [40 Horas]/Aula 05/ContaBanco.java
public class ContaBanco {
    public int numConta;
    protected String tipo; // Cc|Cp
    private String dono;
    private float saldo;
    private boolean status; // Aberta|Fechada

    public ContaBanco() {
        this.setSaldo(0);
        this.setStatus(false);
    }

    public int getNumConta() {
        return numConta;
    }

    public void setNumConta(int numConta) {
        this.numConta = numConta;
    }

    public String getTipo() {
        return tipo;
    }

    public void setTipo(String tipo) {
        this.tipo = tipo;
    }

    public String getDono() {
        return dono;
    }

    public void setDono(String dono) {
        this.dono = dono;
    }

    public float getSaldo() {
        return saldo;
    }

    public void setSaldo(float saldo) {
        this.saldo = saldo;
    }

    public boolean isStatus() {
        return status;
    }

    public void setStatus(boolean status) {
        this.status = status;
    }

    public void abrirConta(String tipo) {
        this.setTipo(tipo);
        this.setStatus(true);
        if ("Cc".equalsIgnoreCase(tipo))
            this.setSaldo(50);
        else if ("Cp".equalsIgnoreCase(tipo))
            this.setSaldo(150);
        System.out.println("Conta aberta com sucesso.");
    }

    public void fecharConta() {
        if (this.getSaldo() > 0)
            System.out.println("Não é possível fechar sua conta, verifique seu saldo.");
        else if (getSaldo() < 0)
            System.out.println("Conta em débito");
        else {
            this.setStatus(false);
            System.out.println("Conta fechada com sucesso");
        }
    }

    public void depositar(float valor) {
        if (this.isStatus()) {
            this.setSaldo(this.getSaldo() + valor);
            System.out.println("Depósito realizado com sucesso na conta de " + this.getDono());
        } else
            System.out.println("Impossível depositar, está conta não está aberta.");
    }

    public void sacar(float valor) {
        if (this.isStatus()) {
            if (this.getSaldo() >= valor)
                this.setSaldo(this.getSaldo() - valor);
            else
                System.out.println("Saldo insuficiente.");
        } else
            System.out.println("Impossível sacar, esta conta está fechada.");
    }

    public void pagarMensal() {
        float valorMensalidade = 0;
        if ("Cc".equalsIgnoreCase(this.getTipo()))
            valorMensalidade = 12;
        if ("Cp".equalsIgnoreCase(this.getTipo()))
            valorMensalidade = 20;
        if (this.isStatus()) {
            if (this.getSaldo() >= valorMensalidade)
                this.setSaldo(this.getSaldo() - valorMensalidade);
            else
                System.out.println("Saldo insuficiente.");
        } else
            System.out.println("Impossível pagar. Conta fechada.");
    }

    public String toString() {
        return ("----------------------------"
                + "\nNum conta: " + this.getNumConta()
                + "\nTipo: " + this.getTipo()
                + "\nDono: " + this.getDono()
                + "\nSaldo: " + this.getSaldo()
                + "\nStatus: " + this.isStatus());
    }

}
